Chap_API_Event.tex

```diff
@@ -105,12 +105,20 @@ \subsection{\code{PMIx_Register_event_handler}}
 
 If the \refarg{cbfunc} is non-\code{NULL}, the function call will be treated as a \emph{non-blocking} call and will return the following:
 
+\returnsimplenb
+The result of the registration operation shall be returned in the provided callback function along with the assigned event handler identifier.
+
+\returnstart
 \begin{itemize}
-\item \refconst{PMIX_SUCCESS} indicating that the request has been accepted for processing and the provided callback function will be executed upon completion of the operation. Note that the library must not invoke the callback function prior to returning from the \ac{API}. The result of the registration operation shall be returned in the provided callback function along with the assigned event handler identifier.
+% TODO: should there be a PMIX_OPERATION_SUCCEEDED here?
+\item \refconst{PMIX_OPERATION_SUCCEEDED}, returned only when \refarg{cbfunc} is non-\code{NULL} and the request was immediately processed successfully - the \refarg{cbfunc} will \textit{not} be called
```

Comment:
No, I'm afraid not - if the request was immediately processed successfully, then the return value will be the non-negative event identifier. You can never get `OPERATION_SUCCEEDED` from this function.